language-server/src/main/java/org/sonarlint/languageserver/SonarLintLanguageServer.java

```diff
@@ -560,7 +567,7 @@ private static URI parseURI(String uriStr) {
 
   // visible for testing
   void analyze(URI uri, String content, boolean shouldFetchServerIssues) {
-    if (!uri.toString().startsWith("file:/")) {
+    if (!Paths.get(uri).toFile().isFile()) {
```

Comment:
This will throw FileSystemNotFoundException if the URI doesn't start with file://

Reply:
@henryju indeed (what was I thinking). Fixed.